### PR TITLE
June Fixes

### LINKED ISSUE #156: Changing data container types does not update available attributes
When a data container type is changed, the available set of attributes on the container is not updated. This is problematic if the user changes between two container types with largely different attributes, like from a `Scalar` to an `Array`. Any attributes not present in the `Scalar` type will be missing and cannot be added manually to the `Array` type.

This needs to be handled by:
1. Removing any attributes that are not available for the new type
2. Adding any attributes the new type has, which are not present in the old type, by filling them with their default values

### LINKED ISSUE #209: The minimap covers up the expand info box button (if visible)

### LINKED ISSUE #220: Troubleshooting Link and Warning When DaCe Daemon Fails
If the dace daemon fails to launch, we should show a clear message about what will and will not work without a daemon. In addition to that, we should show a link that takes you to a troubleshooting page.

### LINKED ISSUE #223: Metadata Not Updated When Opening New SDFG

### PR DESCRIPTION
**Pull request by phschaad**
- [x] First query dace for metadata when opening a new SDFG. Closes #223.
- [x] Improve error messages for failing dace daemon. Closes #220.
- [x] Ensure data container changes lead to correctly setting the new data container attributes. Closes #156.

### CODE REVIEW

File: src/webclients/components/sdfv/utils/attributes_table.ts [hunk 1]
```diff
@@ -38,7 +39,16 @@ import {
 } from './helpers';
 import { ComponentTarget } from '../../../../components/components';
 
-declare const vscode: any;
+function updateAttrTable(): void {
+    // TODO(later): this is an uggly workaround to how the system of filling the
```
Comment (tbennun) [inline src/webclients/components/sdfv/utils/attributes_table.ts:43]:
```suggestion
    // TODO(later): this is an ugly workaround to how the system of filling the
```

File: src/webclients/components/sdfv/utils/attributes_table.ts [hunk 2]
```diff
@@ -1502,6 +1522,73 @@ export function appendDataDescriptorTable(
...
+        const updateNameListener = (prop: KeyProperty) => {
+            // When a data container name is changed, update the data container
+            // and label for all access nodes referencing this data container.
+            const nVal = prop.getValue();
+            if (nVal.valueChanged) {
+                const oldDescriptor = prop.getKey();
+                const newDescriptor = nVal.value;
+
+                doForAllNodeTypes(
+                    sdfg, 'AccessNode', (accessNode: JsonSDFGNode) => {
```
Comment (tbennun) [inline src/webclients/components/sdfv/utils/attributes_table.ts:1534]: I hope this doesn't traverse recursively and change unrelated access nodes?

**Review by tbennun — APPROVED** [review]
One question and one minor change, otherwise LGTM

File: src/webclients/components/sdfv/utils/attributes_table.ts (hunk 2 above)
Comment (phschaad) [reply src/webclients/components/sdfv/utils/attributes_table.ts:1534]: The function would traverse recursively, but the last positional argument (`false`) tells it not to :)

**Pull request merged**